src/lib/Gen/hbs/HandlebarsGenerator.cpp

```diff
@@ -15,13 +15,73 @@
 #include "Builder.hpp"
 #include "MultiPageVisitor.hpp"
 #include "SinglePageVisitor.hpp"
+
+#include "lib/Lib/TagfileWriter.hpp"
+#include "lib/Support/RawOstream.hpp"
+
+#include <llvm/ADT/SmallString.h>
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/Path.h>
+
 #include <mrdocs/Support/Path.hpp>
+
+#include <fstream>
 #include <sstream>
 
 namespace clang {
 namespace mrdocs {
 namespace hbs {
 
+namespace {
+
+/** Generate a Tagfile for cross-referencing. */
+Error
+generateTagfile(
+    HandlebarsCorpus const& hbsCorpus,
+    std::string_view tagFilePath,
+    std::string_view generatedFilesPath,
+    bool isSinglePage)
+{
+    if (tagFilePath.empty())
+        return Error::success();
+
+    auto tagFileWriterEx = TagfileWriter::create(hbsCorpus, tagFilePath, generatedFilesPath, isSinglePage);
+    if (!tagFileWriterEx)
+        return tagFileWriterEx.error();
+
+    auto tagfileWriter = std::move(*tagFileWriterEx);
+    tagfileWriter.initialize();
+    visit(hbsCorpus->globalNamespace(), tagfileWriter);
+    tagfileWriter.finalize();
+    return Error::success();
+}
+
+std::string 
```

Comment:
It would be good to have a comment explaining what `getSinglePageFileName` is intended to do. At least I don't get it anyway.

In part I don't get it because I still don't understand the necessity of `generatedFilesPath` above (why would it need that if all symbols only have anchors).

But putting that aside, if the tagfile writer needs the directory where the files are, then we just need `getParentDir(outputPath);`.

Reply:
The parameter `generatedFilesPath` was renamed to `outputFilesBasePath` to clarify its purpose.

In "multipage" mode, the anchors for symbols come directly from the Corpus, which includes the generated filenames. However, in "singlepage" mode, the anchor is the resulting document (e.g., `reference.adoc`). Since this information is not present in the Corpus, we need to generate it explicitly.

The function `getSinglePageFileName` exists precisely for this reason: to determine the output file name (e.g., .adoc) for use as the anchor in "singlepage" mode.